src/Core/FeatureFlag/FeatureFlagsModifier.php

```diff
@@ -0,0 +1,139 @@
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\FeatureFlag;
+
+use Doctrine\ORM\EntityManagerInterface;
+use InvalidArgumentException;
+use PrestaShop\PrestaShop\Core\Configuration\DataConfigurationInterface;
+use PrestaShopBundle\Entity\FeatureFlag;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * This class is in charge of controlling the state modifications of Feature Flags
+ * through the Back Office interface. It fetches the data needed for the Back Office interface
+ * and handles payloads from the Back Office page, replicating the changes on the data model.
+ */
+class FeatureFlagsModifier implements DataConfigurationInterface
+{
+    /** @var EntityManagerInterface */
+    private $doctrineEntityManager;
+
+    /** @var TranslatorInterface */
+    private $translator;
+
+    /**
+     * @param EntityManagerInterface $doctrineEntityManager
+     * @param TranslatorInterface $translator
+     */
+    public function __construct(EntityManagerInterface $doctrineEntityManager, TranslatorInterface $translator)
+    {
+        $this->doctrineEntityManager = $doctrineEntityManager;
+        $this->translator = $translator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getConfiguration(): array
+    {
+        $configuration = [];
+
+        foreach ($this->getAllFeatureFlags() as $id => $featureFlag) {
+            $configuration[$featureFlag->getName()] = $featureFlag->getState();
+        }
+
+        return $configuration;
+    }
+
+    /**
+     * @return FeatureFlag[] $allFlags
+     */
+    public function getAllFeatureFlags(): array
+    {
+        return $this->doctrineEntityManager->getRepository('PrestaShopBundle:FeatureFlag')->findAll();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function updateConfiguration(array $configuration): array
+    {
+        if (!$this->validateConfiguration($configuration)) {
+            throw new InvalidArgumentException('Invalid feature flag configuration submitted');
+        }
+
+        /** @var array<string, boolean> $configuration */
+        foreach ($configuration as $flagName => $flagState) {
+            $featureFlag = $this->getOneFeatureFlagByName($flagName);
+
+            if ($flagState) {
+                $featureFlag->enable();
+            } else {
+                $featureFlag->disable();
+            }
+        }
+
+        $this->doctrineEntityManager->flush();
+
+        return [];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function validateConfiguration(array $configuration): bool
+    {
+        /** @var array<string, boolean> $configuration */
+        foreach ($configuration as $flagName => $flagState) {
+            if (!is_string($flagName)) {
+                return false;
+            }
+
+            if (!is_bool($flagState)) {
+                return false;
+            }
+
+            if (null === $this->getOneFeatureFlagByName($flagName)) {
```

Comment:
Little suggestion, not mandatory But you end up calling `getOneFeatureFlagByName` twice for each flag:
- one in validateConfiguration
- and a second time in the update loop

You could remove this check from the validation function (which would only validate the format), and check the flag presence during your update loop (by throwing an appropriate exception) Besides since it's checked before the flush it won't save partial flags, which is a good thing in this case

Reply:
Fixed in https://github.com/PrestaShop/PrestaShop/pull/23703/commits/bb2f40a2a10a847f5f1144549532bff524cc3074